src/Razor/src/Microsoft.VisualStudio.LanguageServerClient.Razor/ProjectRazorJsonPublisher.cs

```diff
@@ -131,7 +131,7 @@ internal void ProjectSnapshotManager_Changed(object sender, ProjectChangeEventAr
         {
             // Not currently active, we need to decide if we should become active or if we should no-op.
 
-            if (ProjectSnapshotManager.OpenDocuments.Count > 0)
+            if (!ProjectSnapshotManager.GetOpenDocuments().IsDefaultOrEmpty)
```

Comment:
Consider calling `IsEmpty` and making it a failure to return a default ImmutableArray.